### ISSUE TITLE
 torch :: nn :: conv2d output feature differs from python Model.

### ISSUE THREAD

**Issue by kerry-Cho** [pytorch/pytorch#21332]
Hi. All

I'm implementing mask rcnn in c ++

I found something strange.
It is output differently from the Python model when visualizing the convolution output value

I do not know why.
This seems to be a bug.

Please Check.
The visualization code was used as below.

C++

<code>
void SaveFeatureHeatmap(torch::Tensor x, std::string savename) 
{

	auto heatmap = torch::mean(x, { 1 }, true).squeeze();
	cv::Mat cam = cv::Mat::eye(x.size(2), x.size(3), CV_8UC1);
	heatmap = heatmap.mul(255).clamp(0, 255).to(torch::kU8);
	heatmap = heatmap.to(torch::kCPU);
	std::memcpy((void*)cam.data, heatmap.data_ptr(), sizeof(torch::kU8)*heatmap.numel());
	cv::applyColorMap(cam, cam, cv::COLORMAP_JET);
	cv::resize(cam, cam,cv::Size(512, 512));
	cv::imwrite(savename + ".jpg", cam);
}
</code>

python 

<code>
import torch
from torch import nn

import matplotlib.pyplot as plt

        act = torch.mean(x.detach(),1,True).squeeze()
        fig, axarr = plt.subplots(1)
        axarr.imshow(act)
</code>

inputI mage

![input.jp](https://ifh.cc/g/xdj6H.jpg)

C++  Conv1 Out
![conv1](https://ifh.cc/g/qS9Kc.jpg)

python Conv1 Out
![conv1](https://ifh.cc/g/qXAJf.png)

**Comment by fmassa** [comment]
This is probably a visualization issue.

I'd recommend you check that the values are the same instead of plotting them.

Also, questions are better suited to our discussion forums https://discuss.pytorch.org
